src/message-list/MessageList.js

```diff
@@ -16,17 +19,36 @@ const styles = StyleSheet.create({
     flexGrow: 1,
     justifyContent: 'space-around',
   },
+  container: {
+    flex: 1
+  },
 });
 
 export default class MessageList extends React.PureComponent {
   constructor() {
     super();
     this.state = {
+      shouldShowNewMessageIndicator: false,
       autoScrollToBottom: false,
     };
+
+    // Variables concerned with calculating scroll position
+    this.initialContentHeight = 0;
+    this.initialScrollOffset = 0;
   }
 
   componentWillReceiveProps(nextProps) {
+    // New messages have been received and user has scrolled above the threshold value
+    if (this.hasScrolledBeyondInfoThreshold() && this.didReceiveNewMessages(nextProps)) {
```

Comment:
You don't need to set any state to the component.
Calculate 'shouldShowNewMessageIndicator' in the render function.
Considering my other note about Chat.js, and moving the notification component there, it should be in the render() of Chat.